### PR TITLE
Add smite dialog

### LINKED ISSUE #2663: Create a Smite dialog (UI only)
Create UI for the Smite dialog & unblock/enable it in the actions index Modal.

https://www.figma.com/file/qQoBbyyDXkuHwoqHIOGjEu/Colony-Simplified?node-id=4445%3A2

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

This PR adds in Smite dialog (Just UI)

**New stuff** ✨

* SmiteDialogForm
* SmiteDialog
* TeamDropdownItem

Screenshot 2021-09-07 at 10 40 49

Resolves #2663

### CODE REVIEW

**Review by ArmandoGraterol — APPROVED** [review]
Looks good to me! Nice work :)

File: src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+import styles from './SmiteDialogForm.css';
+
+const MSG = defineMessages({
+  title: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.title',
+    defaultMessage: 'Smite',
+  },
+  team: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.team',
+    defaultMessage: 'Select Team in which punishment should happen',
```
Comment (chinins) [inline src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx:30]: It doesn't sound very correct either. My suggestion would be:
`Select a Team where the punishment should be applied`
Again, I was running it by my native speaker friend.

File: src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  },
+  who: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.who',
+    defaultMessage: 'Pick who is the target',
```
Comment (chinins) [inline src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx:34]: I know that this is the copy from the specs but
1. it doesn't sound very good & grammatically I think it's incorrect - it should be at least `Pick who the target is`
2. it doesn't sound good - too wordy. I ran it by my native speaker friend and he suggested `Select a target` (preferable option) or `Choose a target`.
So I would suggest to change it but it's up to you...

File: src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+  },
+  amount: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.amount',
+    defaultMessage: 'Amount of reputation to deduct',
+  },
+  annotation: {
+    id: 'dashboard.SmiteDialog.SmiteDialogForm.annotation',
+    defaultMessage: 'Explain why you’re making this payment (optional)',
```
Comment (chinins) [inline src/modules/dashboard/components/SmiteDialog/SmiteDialogForm.tsx:42]: I think this is a leftover from the other dialog. It should be:
`Explain why you are smiting the user (optional)`

**Review by chinins — APPROVED** [review]
All good, except an error in one copy and then I have a couple of suggestions for other copies. It's just my suggestions of how it sounds better in English.
